src/geometry/GeoTrackView.hh

```diff
@@ -48,15 +48,39 @@ class GeoTrackView
     // Move to the next boundary
     inline CELER_FUNCTION void move_next_step();
 
+    // Update current volume, called whenever move reaches boundary
+    inline CELER_FUNCTION void move_next_volume();
+
+    // A tiny push to make sure tracks go over boundaries, to minimize chances of getting stuck at boundaries
+    CELER_CONSTEXPR_FUNCTION real_type tolerance()
+    {
+      return 1e-12;
+    }
+
     //@{
     //! State accessors
     CELER_FUNCTION const Real3& pos() const { return pos_; }
     CELER_FUNCTION const Real3&    dir() const { return dir_; }
     CELER_FUNCTION real_type       next_step() const { return next_step_; }
-    inline CELER_FUNCTION VolumeId volume_id() const;
-    inline CELER_FUNCTION Boundary boundary() const;
     //@}
 
+    //@{
+    //! State modifiers via non-const references
+    CELER_FUNCTION Real3& pos() { return pos_; }
+    CELER_FUNCTION Real3& dir() { return dir_; }
+    CELER_FUNCTION real_type& next_step() { return next_step_; }
+    //@}
+
+    /*!
+     * Get the volume ID in the current cell.
+     */
+    inline CELER_FUNCTION VolumeId volume_id() const;
```

Comment:
In header files, try to use one-line comments for functions so that the interface is easier to read (takes up less vertical space).
```suggestion
    //! Get the volume ID in the current cell.
    inline CELER_FUNCTION VolumeId volume_id() const;
```

Reply:
Done, here and also in other header comments in `LinearPropagator.hh`.